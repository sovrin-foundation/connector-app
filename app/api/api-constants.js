// @flow

export const TOKEN_EXPIRED_CODE = 'VES-101'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_REQUEST_ID = 'id'
export const QR_CODE_SENDER_DETAIL = 's'
export const QR_CODE_SENDER_DID = 'd'
export const QR_CODE_SENDER_NAME = 'n'
export const QR_CODE_LOGO_URL = 'l'
export const QR_CODE_SENDER_VERIFICATION_KEY = 'v'
export const QR_CODE_SENDER_KEY_DELEGATION = 'dp'
export const QR_CODE_DELEGATION_DID = 'd'
export const QR_CODE_DELEGATION_KEY = 'k'
export const QR_CODE_DELEGATION_SIGNATURE = 's'
export const QR_CODE_SENDER_AGENCY = 'sa'
export const QR_CODE_SENDER_AGENCY_DID = 'd'
export const QR_CODE_SENDER_AGENCY_KEY = 'v'
export const QR_CODE_SENDER_AGENCY_ENDPOINT = 'e'
export const QR_CODE_TARGET_NAME = 't'
export const GENESIS_FILE_NAME = 'pool_transactions_genesis'

export const MESSAGE_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'credOffer',
  CLAIM_REQUEST: 'credReq',
  PROOF_REQUEST: 'proofReq',
  PROOF: 'proof',
  CLAIM: 'cred',
}

export const PAYLOAD_TYPE = {
  GET_MESSAGE: 'GET_MSGS',
  UPDATE_PUSH_COM_METHOD: 'UPDATE_PUSH_COM_METHOD',
  SEND_MESSAGE: 'SEND_MSG',
}

export const STATUS_CODE = {
  ACCEPT: 'MS-104',
}

export const ERROR_ALREADY_EXIST = {
  code: 'OCS',
  message: 'You already have a relationship with ',
}

export const ERROR_ALREADY_EXIST_TITLE = 'Relationship already established'

export const ERROR_INVITATION_RESPONSE_PARSE_CODE = 'RN-100'

export const ERROR_INVITATION_RESPONSE_PARSE =
  'Error parsing error message in invitation response'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE_CODE = 'RN-101'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE =
  'Error parsing error message Pending Invitation SMS'

export const ERROR_INVITATION_RESPONSE_FAILED =
  'Failed to establish connection. Please try again later.'
