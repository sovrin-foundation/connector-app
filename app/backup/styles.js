// @flow
import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  isBiggerThanShortDevice,
  color,
  grey,
} from '../common/styles/constant'

const { width, height } = Dimensions.get('window')
const SPACE_FILLER = 50
const amountSpacing = isBiggerThanShortDevice ? PADDING_VERTICAL * 2 : 0
const submitButtonHeight = isBiggerThanShortDevice ? 63 : 43
const showRecoveryButtonHeight = isBiggerThanShortDevice ? 150 : 100
const inputBoxHeight =
  isBiggerThanShortDevice || Platform.OS === 'ios' ? 137 : 40
export const chatBubbleTextOffset = 40
const styles = StyleSheet.create({
  tabContainer: {
    maxHeight: 60,
  },
  backgroundImage: {
    width: '100%',
    flex: 1,
    position: 'absolute',
  },
  headerCloseIcon: {
    marginRight: PADDING_HORIZONTAL,
    marginBottom: PADDING_VERTICAL,
    alignSelf: 'flex-end',
  },
  headerIcon: {
    marginRight: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 0,
    alignSelf: 'flex-end',
  },
  headerBackIcon: {
    marginLeft: isBiggerThanShortDevice ? 5 : 0,
    marginTop: isBiggerThanShortDevice ? 15 : 5,
    alignSelf: 'flex-start',
  },
  headerSpacer: {
    height: '100%',
    width: isBiggerThanShortDevice ? SPACE_FILLER : SPACE_FILLER * 3 / 4,
  },
  showRecoveryPassphrase: {
    width: '86%',
    height: showRecoveryButtonHeight,
    borderRadius: 5,
    alignSelf: 'center',
    backgroundColor: color.bg.tertiary.color,
    elevation: 7,
    marginBottom: 60,
  },
  genRecovery: {
    backgroundColor: color.bg.eleventh.color,
  },
  genRecoveryHeader: {
    width: '100%',
  },
  genRecoveryPhraseContainer: {
    position: 'absolute',
    top: 20,
    left: chatBubbleTextOffset / 2,
    paddingHorizontal: '7%',
  },
  genRecoveryPhraseLoadingContainer: {
    padding: 40,
    position: 'absolute',
    top: 0,
  },
  genRecoveryPhrase: {
    textAlign: 'center',
    color: color.bg.eleventh.color,
    fontSize: isBiggerThanShortDevice ? 20 : 18,
    lineHeight: isBiggerThanShortDevice ? 30 : 28,
    fontWeight: '900',
  },
  genRecoveryText: {
    marginBottom: isBiggerThanShortDevice ? 10 : '2%',
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '600',
  },
  genRecoveryMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
    marginBottom: isBiggerThanShortDevice ? 20 : 5,
  },
  genRecoverySecondMessage: {
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: 'bold',
  },
  genRecoverySmallMessage: {
    fontSize: 14,
    lineHeight: 17,
    fontWeight: 'bold',
    marginBottom: isBiggerThanShortDevice ? 15 : '3%',
  },
  verifyMainContainer: {
    flex: 1,
    backgroundColor: color.bg.twelfth.color,
  },
  verifyMainText: {
    paddingHorizontal: 20,
    fontSize: 18,
    lineHeight: 22,
    fontWeight: '500',
    marginTop: isBiggerThanShortDevice ? 40 : 20,
    marginBottom: isBiggerThanShortDevice ? 40 : 20,
  },
  exportBackupMainText: {
    paddingHorizontal: 10,
    fontSize: isBiggerThanShortDevice ? 18 : 16,
    lineHeight: isBiggerThanShortDevice ? 22 : 20,
    fontWeight: '500',
  },
  inputBox: {
    marginBottom: 24,
    marginRight: 20,
    marginLeft: 20,
    height: inputBoxHeight,
    backgroundColor: 'rgba(0,0,0,0.33)',
    color: 'white',
    padding: 10,
    textAlignVertical: 'top',
  },
  submitButton: {
    marginLeft: '2%',
    marginRight: '2%',
    marginBottom: 15,
    borderRadius: 5,
    shadowColor: grey,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowRadius: 5,
    shadowOpacity: 0.3,
    elevation: 7,
    height: isBiggerThanShortDevice ? 53 : 43,
  },
  backgroundImageVerify: {
    flex: 1,
    position: 'absolute',
  },
  exportBackup: {
    backgroundColor: color.bg.thirteenth.color,
  },
  wrapper: {
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  title: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 20 : 10,
    width: '100%',
  },
  exportBackupTitle: {
    fontWeight: '600',
    lineHeight: isBiggerThanShortDevice ? 27 : 22,
    fontSize: isBiggerThanShortDevice ? 22 : 20,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
    marginTop: isBiggerThanShortDevice ? 10 : 0,
    maxWidth: 290,
  },
  exportBackupText: {
    marginBottom: 20,
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 22,
  },
  exportBackupMessage: {
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 18,
  },
  exportBackupFile: {
    fontSize: 15,
    lineHeight: 18,
    fontWeight: '900',
  },
  exportBackupSmallMessage: {
    fontSize: 14,
    maxWidth: 300,
    lineHeight: 17,
    marginBottom: isBiggerThanShortDevice ? 30 : 10,
  },
  backupComplete: {
    backgroundColor: color.bg.fourteenth.color,
  },
  backupCompleteHeader: {
    width: '100%',
  },
  backupCompleteText: {
    fontSize: 18,
    fontWeight: '500',
    lineHeight: 22,
  },
  alignItemsCenter: {
    marginBottom: 6,
    marginLeft: '5%',
    marginRight: '5%',
  },
  verticalSpacing: {
    paddingTop: amountSpacing,
  },
  customButtonColor: {
    fontWeight: '600',
    fontSize: 18,
  },
  lockIconImage: {
    paddingTop: isBiggerThanShortDevice ? 40 : '2%',
  },
  imageIcon: {
    resizeMode: 'stretch',
    width: '100%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
  imageIconEncryptFile: {
    resizeMode: 'contain',
    width: isBiggerThanShortDevice ? '100%' : '80%',
    marginBottom: isBiggerThanShortDevice ? 20 : 0,
  },
})

export default styles
