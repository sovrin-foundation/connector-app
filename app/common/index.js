// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
