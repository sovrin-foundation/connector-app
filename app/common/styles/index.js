// @flow
export * from './constant'
export * from './tab-constants'
