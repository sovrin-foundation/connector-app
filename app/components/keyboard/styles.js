// @flow

import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  isBiggerThanShortDevice,
  deviceHeight,
  MEDIUM_DEVICE,
} from '../../common/styles/constant'
import { PUSH_NOTIFICATION_PERMISSION } from '../../push-notification/type-push-notification'

const ROW_MARGIN = isBiggerThanShortDevice ? 36 : 15
const FIRST_ROW_MARGIN = isBiggerThanShortDevice ? 24 : 8

const ROW_HEIGHT =
  Platform.OS === 'ios' ? 36 : deviceHeight < MEDIUM_DEVICE ? 27 : 36

const FONT_SIZE = isBiggerThanShortDevice ? 28 : 26
export const INPUT_RANGE = [0, 0.5, 1]
export const OUTPUT_RANGE =
  Platform.OS === 'ios'
    ? isBiggerThanShortDevice ? [28, 29, 30] : [26, 30, 34]
    : deviceHeight < MEDIUM_DEVICE ? [28, 30, 32] : [26, 30, 34]

export const styles = StyleSheet.create({
  container: {
    marginHorizontal: 32, // TODO: update style guide to reflect new constant
    marginBottom: 5,
  },
  firstRow: {
    height: ROW_HEIGHT,
    marginTop: FIRST_ROW_MARGIN,
  },
  row: {
    height: ROW_HEIGHT,
    marginTop: ROW_MARGIN,
  },
  number: {
    fontSize: FONT_SIZE,
    fontWeight: '500', // TODO: update style guide to reflect new constant
  },
})
