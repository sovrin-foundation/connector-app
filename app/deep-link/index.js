// @flow
import React, { PureComponent } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import branch from 'react-native-branch'
import { deepLinkData, deepLinkEmpty, deepLinkError } from './deep-link-store'
import type { DeepLinkProps, DeepLinkBundle } from './type-deep-link'

export class DeepLink extends PureComponent<DeepLinkProps, void> {
  onDeepLinkData = (bundle: DeepLinkBundle) => {
    if (bundle.error) {
      this.props.deepLinkError(bundle.error)
    } else if (bundle.params) {
      if (bundle.params['+clicked_branch_link'] === true) {
        // update store with deep link params
        this.props.deepLinkData(bundle.params.t)
      } else {
        // update store that deep link was not clicked
        Object.keys(this.props.tokens).length === 0 &&
          this.props.deepLinkEmpty()
      }
    } else {
      Object.keys(this.props.tokens).length === 0 && this.props.deepLinkEmpty()
    }
  }

  componentDidMount() {
    branch.subscribe(this.onDeepLinkData)
  }

  render() {
    return null
  }
}

const mapStateToProps = state => ({
  tokens: state.deepLink.tokens,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      deepLinkData,
      deepLinkEmpty,
      deepLinkError,
    },
    dispatch
  )

const DeepLinkConnected = connect(mapStateToProps, mapDispatchToProps)(DeepLink)

export default DeepLinkConnected
