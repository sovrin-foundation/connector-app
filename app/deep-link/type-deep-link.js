// @flow
import type { CustomError, ResetAction } from '../common/type-common'

export const DEEP_LINK_ERROR = 'DEEP_LINK_ERROR'
export const DEEP_LINK_DATA = 'DEEP_LINK_DATA'
export const DEEP_LINK_EMPTY = 'DEEP_LINK_EMPTY'
export const DEEP_LINK_PROCESSED = 'DEEP_LINK_PROCESSED'

export const DEEP_LINK_STATUS = {
  NONE: 'NONE',
  PROCESSED: 'PROCESSED',
}

export type DeepLinkDataAction = {
  type: typeof DEEP_LINK_DATA,
  data: string,
}

export type DeepLinkEmptyAction = {
  type: typeof DEEP_LINK_EMPTY,
}

export type DeepLinkProcessedAction = {
  type: typeof DEEP_LINK_PROCESSED,
  data: string,
}

export type DeepLinkErrorAction = {
  type: typeof DEEP_LINK_ERROR,
  error: ?any,
}

export type Token = {
  +status: string,
  +token: string,
  +error: ?CustomError,
}

export type DeepLinkStore = {
  tokens:
    | {
        +[string]: Token,
      }
    | {},
  isLoading: boolean,
  error: ?string,
}

export type DeepLinkAction =
  | DeepLinkDataAction
  | DeepLinkEmptyAction
  | DeepLinkErrorAction
  | ResetAction

export type DeepLinkBundle = {
  error: ?string,
  params: ?{
    ['+clicked_branch_link']: boolean,
    t: string,
  },
}

export type DeepLinkProps = {
  deepLinkData: (token: string) => void,
  deepLinkEmpty: () => void,
  deepLinkError: string => void,
  tokens: {
    [string]: Token,
  },
}
