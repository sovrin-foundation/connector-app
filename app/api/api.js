// @flow
import { api, options } from './api-utils'
import type {
  GetInvitationLinkApiData,
  MessageApiData,
  DownloadInvitationApiData,
  SendAuthenticationResponseApiData,
  EnvironmentDetailUrlDownloaded,
} from './type-api'
import { PAYLOAD_TYPE, MESSAGE_TYPE, STATUS_CODE } from './api-constants'

export const getInvitationLink = ({
  agencyUrl,
  smsToken,
}: GetInvitationLinkApiData) =>
  api(`${agencyUrl}/agency/url-mapper/${smsToken}`, options('GET'))

export const invitationDetailsRequest = ({ url }: DownloadInvitationApiData) =>
  api(url, options('GET'))

export const getAdditionalData = ({ agencyUrl, dataBody }: MessageApiData) =>
  api(`${agencyUrl}/agency/route`, options('POST', dataBody))

export const sendAuthenticationRequest = ({
  data: { identifier, dataBody },
  config: { agencyUrl },
}: SendAuthenticationResponseApiData) =>
  api(`${agencyUrl}/agent/${identifier}/auth`, options('PUT', dataBody), true)

export const downloadEnvironmentDetails = (
  url: string
): Promise<EnvironmentDetailUrlDownloaded> => api(url, options())
