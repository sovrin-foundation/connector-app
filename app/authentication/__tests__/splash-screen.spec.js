// @flow
import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import {
  splashScreenRoute,
  homeRoute,
  waitForInvitationRoute,
  invitationRoute,
} from '../../common/'
import { PUSH_NOTIFICATION_SENT_CODE } from '../../api/api-constants'
import {
  default as connectedSplashScreen,
  SplashScreenView,
} from '../splash-screen'
import { color } from '../../common/styles'
import SplashScreen from 'react-native-splash-screen'
import {
  getNavigation,
  getStore,
  getSmsPendingInvitationOfToken,
  senderDid1,
} from '../../../__mocks__/static-data'
import { DEEP_LINK_STATUS } from '../../deep-link/type-deep-link'

describe('<SplashScreen />', () => {
  function getProps(overrideProps = {}) {
    const getLock = () => {
      const { lock } = getStore().getState()
      return lock
    }
    // TODO: We have to fix the problem in getStore function, we should be just able to pass an object and
    // TODO: it should deep extend default store state. We can't do these many calls for getStore and getState
    // TODO: for any property that we want to override
    const store = getStore({
      ...getStore().getState(),
      lock: { ...getLock(), isAppLocked: true },
    })
    const {
      deepLink,
      config,
      lock,
      smsPendingInvitation,
      eula,
    } = store.getState()
    const props = {
      navigation: getNavigation(),
      deepLink,
      isInitialized: config.isInitialized,
      lock,
      eula,
      smsPendingInvitation,
      addPendingRedirection: jest.fn(),
      getSmsPendingInvitation: jest.fn(),
      safeToDownloadSmsInvitation: jest.fn(),
      deepLinkProcessed: jest.fn(),
      ...overrideProps,
    }

    return { store, props }
  }

  function setup() {
    const { store, props } = getProps()
    const component = renderer.create(<SplashScreenView {...props} />)
    const instance = component.root.instance

    return { store, props, component, instance }
  }

  it('should match snapshot', () => {
    const { component } = setup()
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should add home route to pending redirection if deepLink is empty', () => {
    const { component, props } = setup()
    const { deepLink, addPendingRedirection } = props
    component.update(
      <SplashScreenView
        {...props}
        deepLink={{ ...deepLink, isLoading: false }}
      />
    )
    expect(addPendingRedirection).toHaveBeenCalledWith([
      { routeName: homeRoute },
    ])
  })

  it('should go To homeRoute if deepLink is empty and app was unlocked', () => {
    const { component, props } = setup()
    const { deepLink, navigation, addPendingRedirection, lock } = props
    component.update(
      <SplashScreenView
        {...props}
        lock={{ ...lock, isAppLocked: false }}
        deepLink={{ ...deepLink, isLoading: false }}
      />
    )
    expect(navigation.navigate).toHaveBeenCalledWith(homeRoute)
  })

  it(`should goto 'waitForInvitation' screen and fetch invitation if a deepLink was found and app was unlocked`, () => {
    const { component, props } = setup()
    const { deepLink, getSmsPendingInvitation, lock, navigation } = props
    const updatedDeepLink = {
      ...deepLink,
      isLoading: false,
      tokens: {
        token1: {
          status: DEEP_LINK_STATUS.NONE,
          token: 'token1',
          error: null,
        },
      },
    }
    component.update(
      <SplashScreenView
        {...props}
        deepLink={updatedDeepLink}
        lock={{ ...lock, isAppLocked: false }}
      />
    )
    expect(getSmsPendingInvitation).toHaveBeenCalledWith('token1')
    expect(navigation.navigate).toHaveBeenCalledWith(waitForInvitationRoute)
  })

  it(`should show invitation if invitation is fetched and app is unlocked`, () => {
    const { component, props } = setup()
    const { deepLink, deepLinkProcessed, lock, navigation } = props
    const updatedDeepLink = {
      ...deepLink,
      isLoading: false,
      tokens: {
        '3651947c': {
          status: DEEP_LINK_STATUS.NONE,
          token: '3651947c',
          error: null,
        },
      },
    }
    const smsPendingInvitation = getSmsPendingInvitationOfToken('3651947c')
    component.update(
      <SplashScreenView
        {...props}
        deepLink={updatedDeepLink}
        lock={{ ...lock, isAppLocked: false }}
        smsPendingInvitation={smsPendingInvitation}
      />
    )
    expect(deepLinkProcessed).toHaveBeenCalledWith('3651947c')
    expect(navigation.push).toHaveBeenCalledWith(invitationRoute, {
      senderDID: senderDid1,
      token: '3651947c',
    })
  })

  it(`should add invitation route to pending redirection if invitation is fetched and app is locked`, () => {
    const { component, props } = setup()
    const {
      deepLink,
      deepLinkProcessed,
      addPendingRedirection,
      lock,
      navigation,
    } = props
    const updatedDeepLink = {
      ...deepLink,
      isLoading: false,
      tokens: {
        '3651947c': {
          status: DEEP_LINK_STATUS.NONE,
          token: '3651947c',
          error: null,
        },
      },
    }
    const smsPendingInvitation = getSmsPendingInvitationOfToken('3651947c')
    component.update(
      <SplashScreenView
        {...props}
        deepLink={updatedDeepLink}
        lock={{ ...lock, isAppLocked: true }}
        smsPendingInvitation={smsPendingInvitation}
      />
    )
    expect(deepLinkProcessed).toHaveBeenCalledWith('3651947c')
    expect(addPendingRedirection).toHaveBeenCalledWith([
      {
        routeName: invitationRoute,
        params: { senderDID: senderDid1, token: '3651947c' },
      },
    ])
  })
})
