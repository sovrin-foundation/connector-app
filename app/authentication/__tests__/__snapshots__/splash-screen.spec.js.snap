// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashScreen /> should match snapshot 1`] = `
<View
  center={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    center={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Image
      collapsable={undefined}
      source={1}
      style={
        Object {
          "height": 30,
          "tintColor": "#777777",
          "transform": Array [
            Object {
              "rotate": "0deg",
            },
          ],
          "width": 30,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 17,
          },
          Object {
            "fontFamily": "Lato",
          },
          Object {
            "backgroundColor": "#333333",
            "color": "#ffffff",
          },
          null,
          Object {
            "textAlign": "center",
          },
          Object {
            "backgroundColor": "transparent",
          },
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          Object {
            "color": "#464646",
          },
          Object {
            "fontSize": 17,
            "fontWeight": "500",
            "letterSpacing": -0.42,
            "lineHeight": 21,
            "marginTop": 20,
          },
        ]
      }
    >
      Please waitâ€¦
    </Text>
  </View>
</View>
`;
